homeassistant/components/todoist/calendar.py

```diff
@@ -221,10 +223,26 @@ async def handle_new_task(call: ServiceCall) -> None:
         if project_id is None:
             raise HomeAssistantError(f"Invalid project name '{project_name}'")
 
+        # Optional section within project
+        section_id: str | None = None
+        if SECTION_NAME in call.data:
+            section_name = call.data[SECTION_NAME].lower()
+            sections = await coordinator.async_get_sections(project_id)
+            for section in sections:
+                if section_name == section.name.lower():
+                    section_id = section.id
```

Comment:
Also break out of the for loop when handling the project name above
```suggestion
                    section_id = section.id
                    break
```